developers_chamber/scripts/qa.py

```diff
@@ -0,0 +1,246 @@
+    """
+    def __init__(self, *checks):
+        """
+        Arguments:
+            checks: Instances of QACheck class.
+        """
+        self.checks = checks
+
+    def run(self):
+        print('-----------------------------------------------------------')
+        print_bold(' Quality Assurance Check')
+        print('-----------------------------------------------------------')
+
+        failed_checks = []
+        for i, check in enumerate(self.checks):
+            print_heading('[{}/{}] {}...'.format(i + 1, len(self.checks), check.name))
+            try:
+                check.run()
+            except QAError as ex:
+                failed_checks.append(check)
+                print_error(ex)
+                print(ex.output)
+            else:
+                print_success('OK')
+
+        print('-----------------------------------------------------------')
+
+        if failed_checks:
+            print_error('FAILURE: {} check(s) failed!'.format(len(failed_checks)))
+            exit_code = 1
+        else:
+            print_success('SUCCESS: All QA checks passed!')
+            exit_code = 0
+
+        print('-----------------------------------------------------------')
+        exit(exit_code)
+
+
+# Specific QA Checks
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+class MissingMigrationsQACheck(QACheck):
+    """
+    Checks that make migrations command does not generate new migrations.
+    """
+    name = 'Check missing migrations'
+
+    def _run_check(self):
+        output = self._run_command('pydev make-migration --dry-run')
+        last_line = output.split('\n')[-1]
+        if last_line != 'No changes detected':
+            raise QAError('Found missing migration(s)!', output)
+
+
+class MigrationFilenamesQACheck(QACheck):
+    """
+    Checks that new migrations introduced in the branch have correct names.
+    """
+    name = 'Check migration filenames'
+
+    def _is_migration_file_with_wrong_name(self, path):
+        match = re.search(r'migrations\/([^\/]+)\.py$', path)
+        return bool(match and not re.search(r'^[0-9]{4}$', match.group(1)))
+
+    def _run_check(self):
+        wrong_name_files = []
+        for diff in self._get_diffs():
+            if diff.new_file and self._is_migration_file_with_wrong_name(diff.b_path):
+                wrong_name_files.append(diff.b_path)
+
+        if wrong_name_files:
+            raise QAError('Found wrongly named migration file:', '\n'.join(wrong_name_files))
+
+
+class MissingTranslationsQACheck(QACheck):
+    """
+    Checks that generating translations does not introduce any changes.
+    """
+    name = 'Check missing translations'
+
+    def _is_translation_file(self, path):
+        return bool(re.search(r'django\.po$', path))
+
+    def _run_check(self):
+        self._run_command('pydev make-msgs')
```

Comment:
I am not sure if this will grab translations from all containers (?)

Reply:
This is our custom alias, I surely wouldn't use `make-msgs` here.

Reply:
But every project using QA should have this, this is imo better solution than invoking the command directly, since different project might implement translation generation differently. (e.g. one vs multiple containers) and it does not make sense to me to mess with it here.

Reply:
yes Radim is right. This can be problem. Maybe use some setting for makemessages command definition?

Reply:
If I use `project run` to run message discovery, I need to know structure of the project e.g. which containers to run the command in. This shouldn't be here IMHO. Moreover, if the command invocation used here will be different from that used in local development, discrepancies that will be hard to fix could arise. Why do you think it can be problem? Please suggest me better solution since I cannot think of any.